#Skeleton Key for RNAseq analysis
*Written By: Ciera Martinez*

## About 

This is the script used to perform differential gene expression analysis using `edgeR`.  The tissue came from p4 leaves of *Solanum lycopersicum* using Laser Capture Microdissection. 

## Key to Samples

genotype: either wildtype of *tf2*

region: A. tip B. early emmerging leaflet C. base

type: MBR = Marginal Blastozone Region, other = the rachis or midvein region

## Run this first chunk before rendering knittedn document

*See README.md for more detailed instructions of how to use script*

```{r}
library(edgeR)
library(yaml)


### Set Working Directory
rstudioapi::getActiveDocumentContext
setwd(dirname(rstudioapi::getActiveDocumentContext()$path))

### Read in YAML guide
## This reads in the information in the `de.yml` file which has the two names of the samples you are interested in comparing. 

yamls <- yaml.load_file("de.yml")

## This part assigns your YMAL to a object in R.  This will be used throughout the script to specify which sample types you are comparing. 

sample1 <- yamls$sample1
sample2 <- yamls$sample2

sample1
sample2
```

## To make report

Run the `render()` function below and everything will be run with report at end.

```{r, eval = FALSE}
library(rmarkdown)
render("skeletonDE_sept2017.Rmd", "pdf_document", output_file = paste(sample1,"_",sample2,"_","DE.pdf",sep=""))
```  

Samples:
```{r}
print(sample1)
print(sample2)
```

## Analysis (Acutal start)


### Read in Data
Read in raw count data per gene.

```{r, results='hide'}
counts <- read.delim("../../data/Ciera_coveragebed_counts.txt", row.names = 1)

colnames(counts)
#need to convert NA to 0 counts
counts[is.na(counts)] <- 0

## Get rid of low count libraries "wtbother1.4", "wtbmbr8", "tf2ambr3"
counts <- counts[,-c(36,35,2)]
```

### Subset DE expirement

Start by subsetting the particular treatments which are being compared. This might need to be modified depending on the naming of your samples.  In my case each sample is named by sample and rep number, so the script is identifying any sample with the sample name given in the `de.yml` file. 

```{r}
colnames(counts)
counts1 <- counts[,grep(sample1, colnames(counts), value = TRUE)]
count1Len <- length(colnames(counts1)) #used to specify library group in next step.

counts2 <- counts[,grep(sample2, colnames(counts), value = TRUE)]
count2Len <- length(colnames(counts2)) #used to specify library group in next step. 

counts <- cbind(counts1, counts2)

head(counts)
```

### Add column specifying library Group

Make a vector called group that will be used to make a new column named group to identify library region type. 

```{r}
group <- c(rep(sample1, count1Len), rep(sample2, count2Len))
d <- DGEList(counts = counts, group = group)
```

Check to see if the group column matches your sample name and they are appropriate.

```{r}
d$samples
```

### Differential expression using edgeR

Make sure there is full understanding of each edgeR command being used. The manual is amazing so read it *before* running the DE analysis below  [edgeR manual](http://www.bioconductor.org/packages/release/bioc/vignettes/edgeR/inst/doc/edgeRUsersGuide.pdf). There are many options and they must be set to be appropriate for your analysis. 

```{r}
cpm.d <- cpm(d) #counts per million
d <- d[rowSums(cpm.d > 5) >= 3,] #This might be a line to adjust. It is removing genes with low counts. 
d <- estimateCommonDisp(d,verbose = T) 
d <- calcNormFactors(d)
d <- estimateCommonDisp(d)

DEtest <- exactTest(d, pair = c(sample1, sample2))
head(DEtest$table)
results <- topTags(DEtest, n = Inf)

dim(results$table)
sum(results$table$FDR < .05) # How many are DE genes?
summary(decideTestsDGE(DEtest,p.value = .05))

sig.genes <- rownames(results$table[results$table$FDR < 0.05,]) # outputs just significant gene names

plotSmear(d,de.tags = sig.genes)
```

Subset all the genes with a significant FDR score less than .05.  

```{r}
results.sig <- subset(results$table, results$table$FDR < 0.05)
dim(results.sig)
```

What are the genes that are misexpressed?
For this we need to add some annotation.

Essentially we are merging two annotations files to 1.) only sig genes 2.) all genes
```{r}
annotation1 <- read.delim("../../data/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header = FALSE)  # Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2 <- read.delim("../../data/ITAG2.3_all_Arabidopsis_annotated.tsv")
annotation <- merge(annotation1,annotation2, by = "ITAG")

## Remove the trailing
annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)
annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)

#Making the only significant gene table
results.sig$ITAG <- rownames(results.sig)  #change row.names to ITAG for merging
results.sig.annotated <- merge(results.sig,annotation,by = "ITAG", all.x = TRUE) #This is merging only sig genes

#Making all table
results$table$ITAG <- rownames(results$table)
results.all.annotated <- merge(results$table, annotation,by = "ITAG")
```

Write table with results. 

```{r}
write.table(results.all.annotated, file = paste(sample1, "_",sample2,"_", "DE_all.txt",sep = ""), 
            sep = "\t", row.names = F)
write.table(results.sig.annotated, file = paste(sample1, "_", sample2, "_", "DE_sig.txt", sep = ""), 
            sep = "\t", row.names = F)
```

Now run the script below for a full `knitr` report of what was run and leave this report in the folder that the analysis was done with output files. 



