# Differential Gene Expression of each region seperately 

The purpose of this is to visualize the apparent difference in the number of upregulated genes between the longitudinal axis. 

The first step is to upload all the results of all the DE


```{r}
library(plyr)
library(ggplot2)
library(reshape2)

#The first step is to get a list of the file names in the directory you are interested in. 
filenames <- list.files("onlySig/", pattern = "*.txt", full.names = FALSE)


#This opens all the files and creates a dataframe named after the filename
for (i in 1:length(filenames)) {
    assign(filenames[i], read.delim(paste("onlySig//", filenames[i], sep = "")))  
}


#This puts all the dataframes into a list.  Basically you have a list of dataframes.
mydata <- list()
for (i in seq_along(filenames)) {
      mydata[[i]] <- read.delim(paste("onlySig//", file = filenames[i], sep = ""))
}

# Now I am looping each dataframe and making a column called RNAseq, which contains the name of the file.  I used the original filenames list to loop through. 

for (i in seq_along(mydata)) {
  mydata[[i]]$RNAseq <- rep(filenames[i],nrow(mydata[[i]]))
}

#Tests
head(filenames[1])#Check
head(mydata[[1]]) 

head(filenames[2])#Check
head(mydata[[2]]) 
```
 
Now that I have the list of list with a new column name, I can merge them all together.

```{r}
singleDF <- ldply(mydata, data.frame)
```

Now that I have one nice and neat data frame of my RNAseq results I will begin to summarize so that I can finally visualize.  

### Things to do to data and questions to answer:

[ ] how many upregulated genes vs downregulated genes. 
[ ] make a table
[ ] use the RNAseq tab to categorize your results
[x] get rid of all those columns you do not need

```{r}
# get rid of all those columns you do not need
singleDF$RNAseq <- gsub("_DE_sig.txt", "", singleDF$RNAseq)

#keep only what you need
names(singleDF)
smallSingle <- singleDF[, c(1:5,15)] 
head(smallSingle)

   
#Just exploring
## Better Graph below
ggplot(smallSingle, aes(RNAseq, logFC)) + 
  geom_bar(stat = "identity") +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
  geom_hline(yintercept=0, color = "red", size=.5)
```

```{r}
head(smallSingle)
smallSingle$RNAseq <- as.factor(smallSingle$RNAseq)
levels(smallSingle$RNAseq)

base_tf2 <- smallSingle %>%
  filter(RNAseq == "tf2cmbr_tf2cother")
base_wt <- smallSingle %>%
  filter(RNAseq == "wtcmbr_wtcother")

base_intersects <- as.data.frame(intersect(base_tf2$ITAG, base_wt$ITAG))
base_intersects <- as.data.frame(base_intersects)
colnames(base_intersects) <- "ITAG"

annotation1 <- read.delim("../../../../../../../06diffGeneExp/analysis1_2014/data/ITAG2.3_all_Arabidopsis_ITAG_annotations.tsv", header = FALSE)  #Changed to the SGN human readable annotation
colnames(annotation1) <- c("ITAG", "SGN_annotation")
annotation2 <- read.delim("../../../../../../../06diffGeneExp/analysis1_2014/data/ITAG2.3_all_Arabidopsis_annotated.tsv")

annotation <- merge(annotation1, annotation2, by = "ITAG")

#head(annotation)
colnames(annotation)
head(annotation)
#head(results.sig)

annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)
annotation$ITAG <- gsub("^(.*)[.].*", "\\1", annotation$ITAG)



results.sig.annotated <- merge(base_intersects, annotation, by = "ITAG", all.x = TRUE) #This is merging only sig genes

write.csv(results.sig.annotated, "../../../../../datasets/base_intersects_wtvstf2.csv")
```

```{r}
#Upregulated vs Downregulated 
head(smallSingle)

RNAseqNames <- unique(smallSingle$RNAseq)

#Basically I need to make a loop that extracts how many upregualted genes vs how many down regulated genes.  I need to populate a dataframe that has rnaSeq, #upregulated, #downregulated.
head(smallSingle)
RNAseqNames

downRegulated <- vector('numeric')
for (i in 1:length(filenames)) {
 downRegulated[i] <- nrow(subset(smallSingle, RNAseq == RNAseqNames[i] & logFC < 0))
}

upRegulated <- vector('numeric')
for (i in 1:length(filenames)) {
 upRegulated[i] <- nrow(subset(smallSingle, RNAseq == RNAseqNames[i] & logFC > 0))
}

results <- melt(data.frame(RNAseqNames, upRegulated, downRegulated))
colnames(results) <- c("DE", "regulated","value")

# Change wording to match other figures
results$regulated <- gsub("upRegulated", "rachis", results$regulated)
results$regulated <- gsub("downRegulated", "margin", results$regulated)

results$genotype <- c(rep("tf-2", 3), rep("WT", 3), rep("tf-2", 3), rep("WT", 3))

results$DE2 <- c(rep(c("top", "middle", "base"), 4))

ggplot(results, aes(DE2, value, fill = regulated)) + 
  geom_bar(stat = "identity") + 
  scale_fill_manual(values = c("#bdbec0","#4da2da")) +
  xlab("Comparison Between Margin and Rachis") +
  ylab("# of DE gene") +
  theme_bw(base_size = 25) +
  theme(axis.text.x = element_text(angle = 60, hjust = 1, vjust = 1)) +
  guides(fill=guide_legend(title = "up-regulated in")) +
  facet_grid(genotype~.)
```

