---
output: html_document
editor_options: 
  chunk_output_type: console
---
# Looking and visualizing individual genes: CAB


date: January 17, 201

Purpose: This is to verify the CAB genes expression data after the remapping in July 2017.  This was done previously using and this analysis is based of the `individualGenesCAB.10172016.Rmd` file.


```{r, eval = false}
library(rmarkdown)
library(reshape2)
library(ggplot2)
library(plyr)
library(RColorBrewer)

#need to set palette colors for each the six regions
lcmPalette <- c( "#b2d9a6", "#0f7c3e","#bf9e71", "#956025", "#b3a2ce", "#4753a4")
```

```{r}
# Read in normalized read counts from new mapping done previously.
countData <- read.csv("../../DE_and_GO/analysis3_modelled/data/output/01_normalizedReadCount19July2017.csv")
```

Set up the dataframe for analysis and visualization
```{r}
head(countData)
colnames(countData)
countData <- countData[,-1]
head(countData)

#Melt Data
countData <- melt(countData)
head(countData)
colnames(countData) <- c("gene", "sample", "count")
head(countData)

#set genotype
countData$genotype <- ifelse(grepl("wt", countData$sample, ignore.case = T), "wt", 
         ifelse(grepl("tf2", countData$sample, ignore.case = T), "tf-2", "unknown"))

#set type
countData$tissue <- ifelse(grepl("other", countData$sample, ignore.case = T), " rachis", 
         ifelse(grepl("mbr", countData$sample, ignore.case = T), " MBR", "unknown"))

#Set Region
countData$region <- ifelse(grepl("a", countData$sample, ignore.case = T), "top", 
         ifelse(grepl("c", countData$sample, ignore.case = T), "base", "middle"))

## Check
head(countData)
summary(as.factor(countData$genotype))
```

## Part two
### Finding CAB genes in Tomato

Option 1 is to just look for all genes that are "orthologous" as defined by the annotation sheet. So all that were defined by previous annotation work that defined any tomato gene as AT1G29930 (CAB1) and AT1G29920 (CAB2) 

```{r}
annotation2 <- read.delim("../../06diffGeneExp/analysis4_1Sept2017/data/ITAG2.3_all_Arabidopsis_annotated.tsv")

annotation2$ITAG <- gsub("^(.*)[.].*", "\\1", annotation2$ITAG)
annotation2$ITAG <- gsub("^(.*)[.].*", "\\1", annotation2$ITAG)


annotation2$cab <- ifelse(grepl("AT1G29930", annotation2$AGI, ignore.case = T), "CAB1", 
         ifelse(grepl("AT1G29920", annotation2$gene_name, ignore.case = T), "CAB2", "NA"))

summary(as.factor(annotation2$cab))
annotationCABs_withdef <- subset(annotation2, cab == "CAB1")

annotationCABs <- unique(annotationCABs_withdef$ITAG)
length(annotationCABs)
  
## There are 10 "CAB1s" identified, although, as we see below, only 3 have any reads that mapped. Also, to note, there are no CAB2s found. 

###################
## Testing if there are CABs in the raw counts
raw_counts <- read.table("../../06diffGeneExp/analysis3_19July2017/data/Ciera_coveragebed_counts.txt", sep = "\t", header = TRUE)
colnames(raw_counts)[1] <- "itag"
allCABs_rawCount <- subset(raw_counts, itag %in% annotationCABs)

## There are only 3 of these genes that had any counts in my raw data
## Either there is an annotation problem, or there are no reads from CAB2 genes

#######################

## Find which CABs are significant. 
## [x] 1. in the the modeled DE analysis

modelled <- read.csv("../../06diffGeneExp/analysis3_19July2017/data/output/wt.question1.onlysig_0.05_.txt")
head(modelled)
allCABs_sig_modelled <- subset(modelled, ITAG %in% annotationCABs)


## [x] 2. in the pairwise of each region
## This was made in a file called 'making_All_Files
pairwise <- read.csv("../../06diffGeneExp/analysis4_1Sept2017/analysis/R_afterDEAnalysis/allsig_DE_from_1Sept2017_analysis.csv")
head(pairwise)

allCABs_sig_pairwise <- subset(pairwise, ITAG %in% annotationCABs)
colnames(allCABs_sig_pairwise)

to_view <- allCABs_sig_pairwise[ , c(1,4,5,15)]

## All three CAB1s are differentially expressed each of the top and middle, but not in the base, although the pattern of upregulation in the rachis is apparent (see below)
unique_allCABs_sig_pairwise <- unique(allCABs_sig_pairwise$ITAG)

```

-------------------------------------------
## Specify gene of interest 

Option 2: Sequences producing significant alignments with Arabidopsis CAB 1 : AT1G29930.1 from BLAST in Solgenomics

SubjectId	id%	Aln	evalue	Score	Description
- Solyc02g071010.1 x	81.20	540/665	2e-87	323	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** Q41422_SOLTU) Length = 798
- Solyc02g070980.1 x	80.94	535/661	1e-82	307	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** K4B876_SOLLC) Length = 798
- Solyc02g070950.1 x	80.70	531/658	1e-78	293	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** Q41422_SOLTU) Length = 798
- Solyc02g070990.1 x	80.64	533/661	6e-78	291	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** K4B877_SOLLC) Length = 798
- Solyc02g070940.1 x	80.64	533/661	6e-78	291	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** Q41422_SOLTU) Length = 798
- Solyc03g005760.1 x	80.54	534/663	9e-77	287	chlorophyll a/b-binding protein Length = 804
- Solyc02g070970.1 x	80.48	532/661	1e-75	283	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** Q41422_SOLTU) Length = 798
- Solyc03g005770.3 x 80.09	555/693	5e-72	272	chlorophyll a/b-binding protein Length = 789
- Solyc02g071030.2 x 80.18	530/661	9e-71	268	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** Q41422_SOLTU) Length = 798
- Solyc03g005780.3 x	80.09	531/663	1e-69	264	chlorophyll a/b-binding protein Cab-3C Length = 804
- Solyc02g071000.1 x	80.03	529/661	2e-68	260	Chlorophyll a-b binding protein, chloroplastic (AHRD V3.3 *** K4B878_SOLLC) Length = 798

```{r}
head(countData)
CABs <- c('Solyc02g071010',
          'Solyc02g070980',
          'Solyc02g070950',
          'Solyc02g070990',
          'Solyc02g070940',
          'Solyc03g005760',
          'Solyc02g070970',
          'Solyc03g005770',
          'Solyc02g071030',
          'Solyc03g005780',
          'Solyc02g071000')

allCABs <- subset(countData, gene %in% CABs)
unique(allCABs$gene) # 3/11 are present
## The same three as identified earlier
```

## Visualization of the three CAB genes in tomato

## WT only - Manuscript Image

```{r}
allCABs_normCount$sample2  <- paste(allCABs_normCount$region, allCABs_normCount$tissue, sep = "")

wtallCABs <- subset(allCABs_normCount, genotype == "wt")
wtallCABs$region_f = factor(wtallCABs$region, levels = c('top','middle','base'))

colnames(wtallCABs)[1] <- "gene"

unique(wtallCABs$gene)

## Change Gene name to be clearer to read.
wtallCABs$gene <- gsub("Solyc03g005760", "SlCAB1", wtallCABs$gene) 
wtallCABs$gene <- gsub("Solyc03g005770", "SlCAB2", wtallCABs$gene) 
wtallCABs$gene <- gsub("Solyc03g005780", "SlCAB3", wtallCABs$gene) 

ggplot(wtallCABs, aes(gene, count, fill = sample2)) + 
  geom_boxplot() +
  theme_bw() +
  theme(text = element_text(size = 20),
        axis.text.x = element_text(angle = 60, hjust = 1, face = "bold", size = 30),
        axis.title.y = element_text(size = 30), 
        axis.title.x = element_text(size = 30),
        axis.text.y  = element_text(size = 20),
        legend.position = "none",
        strip.text.x = element_text(size = 30)) +
  labs(x = "", y = "normalized read count") +
  scale_fill_manual(values = lcmPalette) +
  facet_grid(.~region_f)
```


